packages/block-library/src/block/edit.js

```diff
@@ -6,207 +6,158 @@ import { partial } from 'lodash';
-			isFetching,
-			isSaving,
-			canUpdateBlock,
-			settings,
-		} = this.props;
-		const { isEditing, title, blocks } = this.state;
-
-		if ( ! reusableBlock && isFetching ) {
+	if ( ! reusableBlock ) {
+		if ( isFetching ) {
 			return (
 				<Placeholder>
 					<Spinner />
 				</Placeholder>
 			);
 		}
-
-		if ( ! reusableBlock ) {
-			return (
-				<Placeholder>
-					{ __( 'Block has been deleted or is unavailable.' ) }
-				</Placeholder>
-			);
-		}
-
-		let element = (
-			<BlockEditorProvider
-				settings={ settings }
-				value={ blocks }
-				onChange={ this.setBlocks }
-				onInput={ this.setBlocks }
-			>
-				<WritingFlow>
-					<BlockList />
-				</WritingFlow>
-			</BlockEditorProvider>
-		);
-
-		if ( ! isEditing ) {
-			element = <Disabled>{ element }</Disabled>;
-		}
-
 		return (
-			<div className="block-library-block__reusable-block-container">
-				{ ( isSelected || isEditing ) && (
-					<ReusableBlockEditPanel
-						isEditing={ isEditing }
-						title={ title !== null ? title : reusableBlock.title }
-						isSaving={ isSaving && ! reusableBlock.isTemporary }
-						isEditDisabled={ ! canUpdateBlock }
-						onEdit={ this.startEditing }
-						onChangeTitle={ this.setTitle }
-						onSave={ this.save }
-						onCancel={ this.stopEditing }
-					/>
-				) }
-				{ element }
-			</div>
+			<Placeholder>
+				{ __( 'Block has been deleted or is unavailable.' ) }
+			</Placeholder>
 		);
 	}
-}
 
-export default compose( [
-	withSelect( ( select, ownProps ) => {
-		const {
-			__experimentalGetReusableBlock: getReusableBlock,
-			__experimentalIsFetchingReusableBlock: isFetchingReusableBlock,
-			__experimentalIsSavingReusableBlock: isSavingReusableBlock,
-		} = select( 'core/editor' );
-		const { canUser } = select( 'core' );
-		const { __experimentalGetParsedReusableBlock, getSettings } = select(
-			'core/block-editor'
-		);
-		const { ref } = ownProps.attributes;
-		const reusableBlock = getReusableBlock( ref );
-
-		return {
-			reusableBlock,
-			isFetching: isFetchingReusableBlock( ref ),
-			isSaving: isSavingReusableBlock( ref ),
-			blocks: reusableBlock
-				? __experimentalGetParsedReusableBlock( reusableBlock.id )
-				: null,
-			canUpdateBlock:
-				!! reusableBlock &&
-				! reusableBlock.isTemporary &&
-				!! canUser( 'update', 'blocks', ref ),
-			settings: getSettings(),
-		};
-	} ),
-	withDispatch( ( dispatch, ownProps ) => {
-		const {
-			__experimentalFetchReusableBlocks: fetchReusableBlocks,
-			__experimentalUpdateReusableBlock: updateReusableBlock,
-			__experimentalSaveReusableBlock: saveReusableBlock,
-		} = dispatch( 'core/editor' );
-		const { ref } = ownProps.attributes;
-
-		return {
-			fetchReusableBlock: partial( fetchReusableBlocks, ref ),
-			onChange: partial( updateReusableBlock, ref ),
-			onSave: partial( saveReusableBlock, ref ),
-		};
-	} ),
-] )( ReusableBlockEdit );
+	let content = (
+		<BlockEditorProvider
+			settings={ settings }
+			value={ localBlocks }
```

Comment:
Personally, I might have implemented this slightly different to avoid storing two copies of the same blocks array, where `localBlocks` would only be set if editing, then this could be `value={ localBlocks ?? blocks }`

Reply:
That's a good idea. I'm currently working on trying it out locally. It might be a bit tricky to implement depending on how the `onChange` and `onInput` callbacks on `BlockEditorProvider` work, though. I need to make sure the local state remains `null`ed when not editing or else the bugs I fixed will probably become unfixed.

Reply:
I tried to do this in https://github.com/WordPress/gutenberg/pull/21427/commits/f16815a21d4926b4b074bb3e1a961dbc1ec7bab3, but it's causing reusable blocks to simply not load properly at all. Any ideas as to why?

Reply:
> I tried to do this in [f16815a](https://github.com/WordPress/gutenberg/commit/f16815a21d4926b4b074bb3e1a961dbc1ec7bab3), but it's causing reusable blocks to simply not load properly at all. Any ideas as to why?

In the commit, you're providing a function as `value` to `BlockEditorProvider` which, as far as I know, is not supported.

Shouldn't it be:

```js
value={ isEditing ? localBlocks : blocks }
```

Reply:
Thanks! Things are almost fixed now, but there's still one issue: saving a modified reusable block crashes the editor and outputs [React error 185](https://reactjs.org/docs/error-decoder.html/?invariant=185) in the JS console.

So it looks like there's some kind of recursion issue, but I can't figure out where.

Reply:
Alright, this is working now.